app/src/main/java/com/muzima/view/forms/HTMLFormDataStore.java

```diff
@@ -95,6 +95,7 @@ public void saveHTML(String jsonPayload, String status) {
 
     @JavascriptInterface
     public void saveHTML(String jsonPayload, String status, boolean keepFormOpen) {
+        jsonPayload = injectUserSystemIdToEncounterPayload(jsonPayload);
```

Comment:
Shouldn't it do some validation  before injecting the system user ID? e.g to make sure the value is missing before attempting to inject?

Reply:
I didn't think of that. Let me Put the validation before you merge.